### PR TITLE
c-blosc: add pkg_config name and do not use anymore CXX LINKER for Snappy dependency

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **c-blosc/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/c-blosc/all/conanfile.py [hunk 1]
```diff
@@ -45,21 +45,17 @@ def config_options(self):
 
     def configure(self):
         del self.settings.compiler.cppstd
-        # Snappy is a C++ library with C API. C-Blosc only uses the C API of
-        # Snappy, but if C-Blosc is shared (and snappy static), we have to use
-        # C++ linker.
-        if not (self.options.with_snappy and self.options.shared):
-            del self.settings.compiler.libcxx
+        del self.settings.compiler.libcxx
```
Comment (madebr) [inline recipes/c-blosc/all/conanfile.py:48]:
```suggestion
        del self.settings.compiler.libcxx
        del self.settings.compiler.cppstd
```

Comment (SpaceIm) [reply recipes/c-blosc/all/conanfile.py:48]: `self.settings.compiler.cppstd` is already deleted above

Comment (madebr) [reply recipes/c-blosc/all/conanfile.py:48]: Oops indeed!

**Comment by SpaceIm** [comment]
Maybe I could take the opportunity to add an option about simd intrinsics. It's controlled by `DEACTIVATE_SSE2` and `DEACTIVATE_AVX2` which are False (default value) for the moment. It means that with the current recipe, there is no way to disable AVX2, which could be an issue for some CPU.

**Pull request merged**